crates/dbsp/src/circuit/runtime.rs

```diff
@@ -163,6 +277,22 @@ impl Runtime {
         let nworkers = workers.len();
         let runtime = Self(Arc::new(RuntimeInner::new(layout)));
 
+        // Install custom panic hook.
+
+        // The first call to `take_hook` should clear any custom hooks installed by
```

Comment:
first or last?

Reply:
First. This function clears any user-installed hooks and replaces them with a default hook instead. The second call will return the system hook.